src/Commands/Site/NewSiteLevelFileVersionBatchDeleteJob.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.SharePoint.Client;
+
+using PnP.Framework.Utilities;
+using PnP.PowerShell.Commands.Base.PipeBinds;
+
+using System.Management.Automation;
+
+using Resources = PnP.PowerShell.Commands.Properties.Resources;
+
+namespace PnP.PowerShell.Commands.Sites
+{
+    [Cmdlet(VerbsCommon.New, "SiteLevelFileVersionBatchDeleteJob")]
+    public class NewSiteLevelFileVersionBatchDeleteJob : PnPSharePointCmdlet
+    {
+        [Parameter(Mandatory = true)]
+        public int DeleteBeforeDays;
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force;
+
+        protected override void ExecuteCmdlet()
+        {
+            if (Force || ShouldContinue("By executing this command, versions specified will be permanently deleted. These versions cannot be restored from the recycle bin. Are you sure you want to continue?", Resources.Confirm))
+            {
+                var site = ClientContext.Site;
+                site.StartDeleteFileVersions(DeleteBeforeDays);
```

Comment:
Should there be a check here if the DeleteBeforeDays falls within a certain range? I.e. what happens if 0 or a negative number would be provided?

Reply:
This is checked server-side: 
<img width="834" alt="image" src="https://github.com/pnp/powershell/assets/129248125/521f98f6-ee99-4706-8410-114ab6d81aeb">